### PR TITLE
[WIP] vkconfig: add layers manifests validation against schema

### PR DESCRIPTION
**Pull request by christophe-lunarg**
Validate generated layer manifest against the layer schema

### CODE REVIEW

File: layersvt/VkLayer_device_simulation.json.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-    "file_format_version" : "1.1.0",
-    "layer" : {
+    "file_format_version" : "1.4.0",
```
Comment (jeremyk-lunarg) [inline layersvt/VkLayer_device_simulation.json.in:2]: The section on [Layer Manifest File Format](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/loader/LoaderAndLayerInterface.md#layer-manifest-file-format) in [Loader And Layer Interface](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/loader/LoaderAndLayerInterface.md) states the following on the "file_format_version":
> Supported versions are: 1.0.0, 1.0.1, 1.1.0, 1.1.1, and 1.1.2.


How did you get this version number?

**Review by jeremyk-lunarg — COMMENTED** [review]
You're adding a lot of stuff to the layer manifest file and the file format version is not supported.

Where's the documentation for these new keywords such as "presets"?

File: layersvt/VkLayer_device_simulation.json.in (hunk 1 above)
Comment (christophe-lunarg) [reply layersvt/VkLayer_device_simulation.json.in:2]: Looking at all our layers I found some with version 1.3 (gfxrecon and/or validation layers) so I bumped the version to 1.4 due to the new features. I understand that this doc is not up to date...

**Comment by christophe-lunarg** [comment]
> You're adding a lot of stuff to the layer manifest file and the file format version is not supported.
> 
> Where's the documentation for these new keywords such as "presets"?

Currently, it's undocumented beside the JSON schema itself because it's only for internal use (just LunarG) but the plan is to work on this for next version and make is public.

**Comment by jeremyk-lunarg** [comment]
Might be good to get @lenny-lunarg to take a look at some of these changes.

**Pull request closed without merging**